internal/resolver/dns/dns_resolver_test.go

```diff
@@ -1290,3 +1290,55 @@ func (s) TestMinResolutionInterval(t *testing.T) {
 		r.ResolveNow(resolver.ResolveNowOptions{})
 	}
 }
+
+// TestMinResolutionInterval_NoExtraDelay verifies that there is no extra delay
+// between two resolution requests apart from [MinResolutionInterval]
+// i.e. if a request is made after [MinResolutionInterval], it should resolve
+// immediately. Test sets [MinResolutionInterval] to 1 second and calls
+// ResolveNow() 4 times with a wait of 1 second between each request.
+// A correct implementation should resolve all requests without timing out in
+// test duration of 5 seconds.
+func (s) TestMinResolutionInterval_NoExtraDelay(t *testing.T) {
+	const target = "foo.bar.com"
+
+	overrideResolutionInterval(t, 1*time.Second)
+	tr := &testNetResolver{
+		hostLookupTable: map[string][]string{
+			"foo.bar.com": {"1.2.3.4", "5.6.7.8"},
+		},
+		txtLookupTable: map[string][]string{
+			"_grpc_config.foo.bar.com": txtRecordServiceConfig(txtRecordGood),
+		},
+	}
+	overrideNetResolver(t, tr)
+
+	r, stateCh, _ := buildResolverWithTestClientConn(t, target)
+
+	wantAddrs := []resolver.Address{{Addr: "1.2.3.4" + colonDefaultPort}, {Addr: "5.6.7.8" + colonDefaultPort}}
+	wantSC := scJSON
+
+	// set context timeout to test duration of 5 seconds
+	// to make sure all 4 resolutions happen successfully
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
```

Comment:
Use `defaultTestTimeout` instead of hard coding to `5s`. Also, get rid of the comment.